### DISCUSSION TITLE
Recognizing animals as entities

### DISCUSSION THREAD

**Discussion by abdulghaffarnba** [explosion/spaCy#5900] [Help: Other Questions] [answered]
We are a company in the veterinary industry where we develop online booking platforms for the veterinary clinics. We are going to integrate entity recognition into our services via Spacy as Spacy is known to be the best and fastest entity recognition service available in the industry.

Spacy has three static model for the pipeline as en_core_web_sm, en_core_web_md and en_core_web_lg, which helps to recognize the entities from smaller phrases to large complex text files in an instance of time.

import spacy
nlp = spacy.load('en_core_web_lg')doc = nlp("My name is sam and I am from New York")
#name entity recognitionfor ent in doc.ents:    print(ent.text,ent.label_)

OUTPUT : 
sam PERSON
New York GPE

The above code actually works perfectly, which helps to identify or recognize the common entities as PERSON, ORGANIZATIONS, LOCATIONS etc. As I have mentioned above we are into the veterinary industry where we have have to especially recognize the pets, but any of the static models doesn't recognize the pets; for example ANIMALS, SPECIES, etc.

import spacy
nlp = spacy.load('en_core_web_lg')doc = nlp("i have a cat")
#name entity recognitionfor 
ent in doc.ents:    
print(ent.text,ent.label_)

I believe "Cat" / "Dog" are not very rare entities to be identified, as they are very general and common.

Why are these common entities not identified?

How can we proceed with this to achieve our goal?

**Comment by honnibal** [comment, marked as answer]
The term "named entity" has a particular sort of definition in the NLP community: https://en.wikipedia.org/wiki/Named_entity . The names of animals don't meet this definition, and so we don't provide a pretrained model that recognises animals. There are actually a potentially unlimited number of categories different users might want to recognise, so we don't try to meet all these various use-cases in the core library.

You could consider annotating text to train a new model for entity recognition. Our annotation tool [Prodigy](https://prodi.gy) is one way to do that. You could use a number of other annotation tools instead also. While we've tried to make Prodigy easy to use, training a new model will always benefit from some understanding of how the models work and how to run machine learning experiments, so if you haven't done this sort of thing before, you might find it helpful to engage a consultant who can help you.
